amundsen_application/static/js/components/common/ResourceList/index.tsx

```diff
@@ -49,40 +58,70 @@ class ResourceList extends React.Component<ResourceListProps, ResourceListState>
     }
   };
 
+  onViewAllToggle = () => {
+    this.setState({ isExpanded: !this.state.isExpanded })
+  };
 
   render() {
-    const { allItems, slicedItems, itemsPerPage, paginate, source } = this.props;
+    const { allItems, customEmptyText, customFooterText, slicedItems, itemsPerPage, paginate, source, title } = this.props;
     const activePage = this.props.activePage !== undefined ? this.props.activePage : this.state.activePage;
     const itemsCount = this.props.slicedItemsCount || allItems.length;
     const startIndex = itemsPerPage * activePage;
 
     let itemsToRender = slicedItems || allItems;
-    if (paginate && allItems) {
+    if ((paginate && allItems) || (!this.state.isExpanded && allItems)) {
       itemsToRender = allItems.slice(startIndex, startIndex + itemsPerPage);
     }
 
     return (
-      <>
-        <ul className="list-group">
-          {
-            itemsToRender.map((item, idx) => {
-              const logging = { source, index: startIndex + idx };
-              return <ResourceListItem item={ item } logging={ logging } key={ idx } />;
-            })
-          }
-        </ul>
+      <div className="resource-list">
+        {
+          title &&
+          <div className="resource-list-title title-3">{title}</div>
+        }
+        {
+          itemsCount === 0 && customEmptyText &&
+          <div className="empty-message body-placeholder">
+            { customEmptyText }
+          </div>
+        }
         {
-          paginate &&
-          itemsCount > itemsPerPage &&
-          <Pagination
-            activePage={ activePage + 1 }
-            itemsCountPerPage={ itemsPerPage }
-            totalItemsCount={ itemsCount }
-            pageRangeDisplayed={ PAGINATION_PAGE_RANGE }
-            onChange={ this.onPagination }
-          />
+          itemsCount > 0 &&
+          <>
+            <ul className="list-group">
+              {
+                itemsToRender.map((item, idx) => {
+                  const logging = { source, index: startIndex + idx };
+                  return <ResourceListItem item={ item } logging={ logging } key={ idx } />;
+                })
+              }
+            </ul>
+            {
```

Comment:
Also on the fence about this, but it might be worth considering building a separate component for the paginated version and the click to expand version. If the bookmarks on the home page are changed to "click to expand" then we can probably deprecate the paginated version of this.

Reply:
The paginated version is still used on `SearchPage`. 

I'm open to splitting them as well because this component is pretty much at the limit of how much branching logic I'd want to commit to maintaining. But I'm also inclined to accept this the way it is, under the expectation that we won't have any new concepts introduced in this area any time soon. 

I will clean up the render logic by splitting it into helper methods. This can at least make things a bit more intuitive to either maintain or copy-paste all this added logic into its own component.